geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java

```diff
@@ -360,6 +360,13 @@ private void processRequest(final Socket sock) {
           versionOrdinal = (short) GOSSIP_TO_GEMFIRE_VERSION_MAP.get(gossipVersion);
         } else {
           // Close the socket. We can not accept requests from a newer version
+          try {
+            sock.getOutputStream().write("unknown protocol version".getBytes());
```

Comment:
This situation should never happen.  When we fetch the protocol version of a Locator we use v5.7.  Then we use the locator's protocol version in future communications.  We will never send a higher version number than the locator knows how to handle.

For SSL, this text is nothing like the response to a HelloClient packet.  On machines I've tested on it results in either a SSLHandshakeException or a SSLException hinting that the other end might be using plain text sockets.